### PR TITLE
LPS-166856 The fragments and widgets search form should give feedback reading with screen reader

### PR DESCRIPTION
**Pull request by veroglez**
This PR depends on this pull https://github.com/liferay-echo/liferay-portal/pull/10067

**Just review the last 5 commits**

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by veroglez** [comment]
ci:test:sf

**Comment by veroglez** [comment]
ci:test:relevant

**Comment by veroglez** [comment]
ci:test:relevant

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/fragments-widgets/components/FragmentsSidebar.js [hunk 1]
```diff
@@ -204,6 +205,12 @@ export default function FragmentsSidebar() {
 	const displayStyleButtonDisabled =
 		searchValue || activeTabId === COLLECTION_IDS.widgets;
 
+	const numberOfResults = useMemo(() => {
+		if (!isNullOrUndefined(searchValue)) {
+			return filteredTabs.flatMap((tab) => tab.collections).length;
+		}
+	}, [filteredTabs, searchValue]);
```
Comment (victorg1991) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/fragments-widgets/components/FragmentsSidebar.js:212]: Can we return null here to make it explicit that when we don't have searchvalue the numberOfResults is null? The same for the one in contents

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/common/components/SearchResultsMessage.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+							Liferay.Language.get('showing-x-results'),
+							numberOfResults
+					  )
+					: Liferay.Language.get('no-results-found');
+
+				setText(message);
+			}, 500);
+		}
+
+		return () => {
```
Comment (victorg1991) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/common/components/SearchResultsMessage.js:40]: We can put the return inside the if so we don't have to declare a variable and set it to null


```javascript
		if (!isNullOrUndefined(numberOfResults)) {
			const timeout = setTimeout(() => {
				const message = numberOfResults
					? sub(
							Liferay.Language.get('showing-x-results'),
							numberOfResults
					  )
					: Liferay.Language.get('no-results-found');

				setText(message);
			}, 500);

                         return () => {
                             clearTimeout(timeout);
		         };
		}

		
```

**Comment by victorg1991** [comment]
Just a couple of suggestions, the rest is great! 💯

**Comment by victorg1991** [comment]
ci:forward

**Comment by ealonso** [comment]
Just started reviewing :)

**Comment by ealonso** [comment]
see: https://github.com/brianchandotcom/liferay-portal/pull/125699

**Pull request closed without merging**